### PR TITLE
fix(notification): dont group single host

### PR DESCRIPTION
**Pull request by EdJoPaTo**
The front of all the URLs has the host anyway, and it's cleaner to read when multiple on a single host change.
Especially when some repos only check a single host.

To simplify the code we could just always create an empty line between different hosts, but that would end up long for many changes on different hosts.

@Teufelchen1 interested in a review?

### CODE REVIEW

**Comment by EdJoPaTo** [comment]
When this plaintext is tried to be rendered as Markdown it will end up being somewhat useless as Markdown will join them together again then. The hostname in between prevented that.

Telegram does not do this. Matrix or Slack might?

File: src/notification.rs [hunk 1]
```diff
@@ -106,21 +107,37 @@ mod change_lines_tests {
     }
 
     #[test]
-    fn same_host() {
+    fn multiple_on_single_host() {
```
Comment (Teufelchen1) [inline src/notification.rs:110]: Nit: Multiple what? Elephants? 🐘
```suggestion
    /// If multiple changes are detected on a single host, then they are written one after the other. 
    fn multiple_change_notifications_on_single_host() {
```

File: src/notification.rs [hunk 2]
```diff
@@ -106,21 +107,37 @@ mod change_lines_tests {
...
             &["https://edjopato.de/", "https://edjopato.de/post/"],
-            "edjopato.de
-- https://edjopato.de/
+            "- https://edjopato.de/
 - https://edjopato.de/post/",
         );
     }
 
+    #[test]
+    fn multiple_per_host() {
```
Comment (Teufelchen1) [inline src/notification.rs:119]:
```suggestion
    /// If multiple changes are detected on multiple hosts, then they are grouped by host and separated by a newline.
    fn two_hosts_with_multiple_changes_each() {
```

**Review by Teufelchen1 — APPROVED** [review]
Looks good! Only two nitpicks regarding my pedantic love for good tests.

File: src/notification.rs (hunk 1 above)
Comment (EdJoPaTo) [reply src/notification.rs:110]: Tests are grouped in modules for this purpose.

https://github.com/EdJoPaTo/website-stalker/blob/b2ed3ecf9507006d7ec0691f98519712af9e3552/src/notification.rs#L89

(partial) output from `cargo test`:

```plaintext
test notification::change_lines_tests::different_hosts ... ok
test notification::change_lines_tests::mixed ... ok
test notification::change_lines_tests::multiple_on_single_host ... ok
test notification::change_lines_tests::multiple_per_host ... ok
test notification::change_lines_tests::single ... ok
test notification::commit_template_tests::empty_commit_always_empty ... ok
test notification::commit_template_tests::prefix ... ok
test notification::commit_template_tests::replace ... ok
test notification::e2e_with_commit ... ok
test notification::e2e_without_commit ... ok
```

Comment (Teufelchen1) [reply src/notification.rs:110]: Oh yes! Of course. Still thinking in C language - where there is not go-to testing and everyone does something different :clown_face:

**Pull request merged**